### DISCUSSION TITLE
Missing features

### DISCUSSION THREAD

**Discussion by itsbalamurali** [SeaQL/sea-orm#620] [Ideas]
Thank you 🙏 for this amazing project & libraries.

The following are the features I've felt missing while integrating using postgres.

1. `Upsert` (Update or Insert).
1. Support for array types (`TEXT[]`).
1. JsonBinary / Json (with or without Option<>) column type generation in code gen.
1. `Count` method.
1. `InsertMany` - handle if trying to insert more than 65535 bind paraments. [See here](https://github.com/launchbadge/sqlx/issues/294#issuecomment-886080306)
1. `json` / `jsonb` select query examples.

**Comment by tyt2y3** [comment]
For 1 & 2, I was aware of attempts of WIP but they are not completed to a state where it'd readily be merged.
For 4, I think you can use the Paginator helper struct (it has https://docs.rs/sea-orm/latest/sea_orm/struct.Paginator.html#method.num_items)
For 5, I don't know what'd be the ideal solution
For 6, I think there are some test cases in our repo

**Comment by billy1624** [comment]
> 6. `json` / `jsonb` select query examples.

Do you mean applying db filters (conditions) on JSON column?
